base/applications/rapps/loaddlg.cpp

```diff
@@ -533,6 +534,7 @@ CDownloadManager::DownloadDlgProc(HWND Dlg, UINT uMsg, WPARAM wParam, LPARAM lPa
 
```

Comment:
Question: if any of the hListView or the Thread creation above fail, the `g_Busy` variable stays with its increased value? Isn't this a bug (unbalanced state. If the WM_INITDIALOG is made to fail, the WM_DESTROY below won't be invoked.)

Reply:
I'm pretty sure you get WM_DESTROY even if WM_INITDIALOG fails. WM_INITDIALOG is the dialog manager and higher level than basic HWND handling.

Reply:
I would invite you to quickly confirm this to me (very simple, with step-by-step debugger force the code flow to go through the return FALSE in the WM_INITDIALOG, and put also a breakpoint in the WM_DESTROY to see whether it's hit)

Reply:
> return FALSE in the WM_INITDIALOG

I think you are confusing this with WM_CREATE now. The [WM_INITDIALOG return value](https://learn.microsoft.com/en-us/windows/win32/dlgbox/wm-initdialog#return-value) just controls the focus...

Reply:
Oh true indeed; to really fail a WM_INITDIALOG one would have to call the EndDialog() function.